backend/src/API/teamEventsAPI.ts

```diff
@@ -8,53 +8,54 @@ const teamEventAttendanceDao = new TeamEventAttendanceDao();
 export const getAllTeamEvents = async (user: IdolMember): Promise<TeamEvent[]> => {
   const canEditTeamEvents = await PermissionsManager.canEditTeamEvent(user);
   if (!canEditTeamEvents) throw new PermissionError('does not have permissions');
-  return TeamEventsDao.getAllTeamEvents();
+  const teamEvents = await TeamEventsDao.getAllTeamEvents();
+  return Promise.all(
+    teamEvents.map(async (event) => ({
+      ...event,
+      attendees: await teamEventAttendanceDao.getTeamEventAttendanceByEventId(event.uuid),
+      requests: []
```

Comment:
`requests` needs to be populated with all pending requests too

Reply:
`attendees` should only be `pending === false`

Although now that I think of it, these two lists could be consolidated into one. Doesn't necessarily have to be done now though, can be a change down the road.